### PR TITLE
Add list of interrupts to PLIC documentation

### PR DESCRIPTION
**Pull request by marnovandermaas**
Closes https://github.com/lowRISC/sonata-system/issues/5

### CODE REVIEW

File: doc/ip/plic.md [hunk 1]
```diff
@@ -5,9 +5,35 @@ Please find more details in [the official specification](https://github.com/risc
...
+| 23, 38 | I2C (0, 1)  | SDA interference |
+| 24, 39 | I2C (0, 1)  | Stretch timeout |
+| 25, 40 | I2C (0, 1)  | SDA unstable |
+| 26, 41 | I2C (0, 1)  | Command complete |
+| 27, 42 | I2C (0, 1)  | Transmit stretch |
+| 28, 43 | I2C (0, 1)  | Transmit threshold |
+| 29, 44 | I2C (0, 1)  | Acquire FIFO full |
+| 30, 45 | I2C (0, 1)  | Unexpected stop |
+| 31, 46 | I2C (0, 1)  | Host timeout |
+| 47     | Ethernet    | Packet ready to read via the SPI connected to the ethernet chip |
```
Comment (HU90m) [inline doc/ip/plic.md:38]: nit: If you're not aligning, I think it looks neater to just remove the trailing `|`.

```suggestion
| 47     | Ethernet    | Packet ready to read via the SPI connected to the ethernet chip
```

File: doc/ip/plic.md [hunk 2]
```diff
@@ -5,9 +5,35 @@ Please find more details in [the official specification](https://github.com/risc
...
-In terms of interrupts, the exact mapping of sources to interrupts will be published here at a later point.
-There are a number of considerations.
+The following table shows the current list of interrupts that are fed through the PLIC.
+For more detailed descriptions of what each interrupt means, please refer to the documentation of each individual hardware IP block.
+More interrupts may be added to this table in the future.
+
+| Number | Block       | Interrupt description |
+|--------|-------------|-----------------------|
+|  0     | None        | Tied to zero |
+|  1,  9 | UART (0, 1) | Transmit watermark |
```
Comment (HU90m) [inline doc/ip/plic.md:15]: nit: This is more readable for me.

```suggestion
|  1,  9 | UART 0, 1 | Transmit watermark |
```

**Review by HU90m — APPROVED** [review]
Thanks Marno

File: doc/ip/plic.md (hunk 1 above)
Comment (marnovandermaas) [reply doc/ip/plic.md]: Agreed, removed those.

File: doc/ip/plic.md (hunk 2 above)
Comment (marnovandermaas) [reply doc/ip/plic.md]: Agreed, removed parentheses.

File: doc/ip/plic.md [hunk 3]
```diff
@@ -5,9 +5,35 @@ Please find more details in [the official specification](https://github.com/risc
...
+| 23, 38 | I2C 0, 1  | SDA interference
+| 24, 39 | I2C 0, 1  | Stretch timeout
+| 25, 40 | I2C 0, 1  | SDA unstable
+| 26, 41 | I2C 0, 1  | Command complete
+| 27, 42 | I2C 0, 1  | Transmit stretch
+| 28, 43 | I2C 0, 1  | Transmit threshold
+| 29, 44 | I2C 0, 1  | Acquire FIFO full
+| 30, 45 | I2C 0, 1  | Unexpected stop
+| 31, 46 | I2C 0, 1  | Host timeout
+| 47     | Ethernet  | Packet ready to read via the SPI connected to the ethernet chip
```
Comment (GregAC) [inline doc/ip/plic.md:38]: Does the ethernet chip really only have a single interrupt source? I would assume there are multiple things that can cause an interrupt, so a more generic 'Interrupt from external SPI ethernet chip' would be a more suitable description.

Comment (marnovandermaas) [reply doc/ip/plic.md:38]: So there is only one interrupt pin from the KSZ8851SNLI-TR chip, but you're right that the description should be more generic as the type of interrupt is specified in the interrupt status register.

Comment (marnovandermaas) [reply doc/ip/plic.md:38]: I've made a PR with an update: https://github.com/lowRISC/sonata-system/pull/144

**Pull request merged**